File: games/planetization/bots/python3/my_bot.py
import asyncio
import random

from core.bot import Bot
from core.networking_client import connect
from core.enums import *


# Example Python3 bot implementation for Planetization game.
class MyBot(Bot):

    # Called only once before the match starts. It holds the
    # data that you may need to setup your bot.
    def setup(self, initial_data):
        self.data = initial_data

    # Called repeatedly while the match is generating. Read
    # game state using state parameter and issue your
    # commands with response parameter.
    def update(self, state, response):

        # Iterate through all the planets you own
        for planet in state["yourPlanets"]:

            # If the planet has enough resources, spawn new unit on it
            if planet["canSpawnNewUnit"]:
                unit_type = UnitType.WORKER if (random.random() < 0.5) else UnitType.WARRIOR
                response.spawn_unit(planet["id"], unit_type)

            max_active_workers = self.data["maxActiveWorkersPerPlanet"]
            num_bots_on_planet = len(planet["idsOfUnitsOnPlanet"])

            # Only a certain number of workers can mine resources on
            # a planet. Let's send the rest of them somewhere else.
            for i in range(max_active_workers, num_bots_on_planet):
                unit_id = planet["idsOfUnitsOnPlanet"][i]

                # Randomly select if you will send the unit to a free or opponent planet
                planets = state["freePlanets"] if random.random() < 0.5 else state["opponentPlanets"]

                if len(planets) > 0:
                    # Select a random planet in planets and send your unit there
                    destination_planet_index = random.randint(0, len(planets) - 1)
                    destination_planet_id = planets[destination_planet_index]["id"]
                    response.send_unit(unit_id, destination_planet_id)


# Connects your bot to match generator, don't change it.
if __name__ == "__main__":
    asyncio.get_event_loop().run_until_complete(connect(MyBot()))

File: games/planetization/bots/kotlin/src/MyBot.kt
import core.Bot
import core.NetworkingClient
import core.api.InitialData
import core.api.MatchState
import core.api.Response
import core.api.UnitType
import java.util.*

/** Example Kotlin bot implementation for Planetization game. */
class MyBot : Bot {

    var rand = Random(12)
    lateinit var data: InitialData;

    // Called only once before the match starts. It holds the
    // data that you may need to setup your bot.
    override fun setup(data: InitialData) {
        this.data = data
    }

    // Called repeatedly while the match is generating. Read game state using
    // state parameter and issue your commands with response parameter.
    override fun update(state: MatchState, response: Response) {

        // Iterate through all the planets you own
        for (planet in state.yourPlanets) {

            // If the planet has enough resources, spawn new unit on it
            if (planet.canSpawnNewUnit) {
                val type = if (rand.nextFloat() < 0.5f) UnitType.WARRIOR else UnitType.WORKER
                response.spawnUnit(planet.id, type)
            }

            // Only a certain number of workers can mine resources on
            // a planet. Let's send the rest of them somewhere else.
            for (i in data.maxActiveWorkersPerPlanet until planet.idsOfUnitsOnPlanet.size) {
                val unitId = planet.idsOfUnitsOnPlanet[i]

                // Randomly select if you will send the unit to a free or opponent planet
                val planets = if (rand.nextFloat() < 0.5f) state.freePlanets else state.opponentPlanets

                if (planets.isNotEmpty()) {
                    // Select a random planet in planets and send your unit there
                    val destinationPlanetIndex = (rand.nextFloat() * planets.size).toInt()
                    val destinationPlanetId = planets[destinationPlanetIndex].id
                    response.sendUnit(unitId, destinationPlanetId)
                }
            }
        }
    }

    // Connects your bot to match generator, don't change it.
    companion object {
        @JvmStatic fun main(args: Array<String>) {
            NetworkingClient.connectNew(args, MyBot())
        }
    }
}
